elements/lisk-chain/src/chain.ts

```diff
@@ -399,45 +336,39 @@ export class Chain {
 		);
 	}
 
-	public async verify(
-		block: Block,
-		_: StateStore,
-		{ skipExistingCheck }: { readonly skipExistingCheck: boolean },
-	): Promise<void> {
+	public async verify(block: Block, _: StateStore): Promise<void> {
 		verifyPreviousBlockId(block, this._lastBlock, this._genesisBlock);
 		validateBlockSlot(block, this._lastBlock, this.slots);
-		if (!skipExistingCheck) {
-			await verifyBlockNotExists(this.dataAccess, block);
-			const transactionsResponses = await checkPersistedTransactions(
-				this.dataAccess,
-			)(block.payload);
-			const invalidPersistedResponse = transactionsResponses.find(
+		await this.blocksVerify.checkTransactions(block);
+	}
+
+	// eslint-disable-next-line class-methods-use-this
+	public async apply(block: Block, stateStore: StateStore): Promise<void> {
+		if (block.payload.length > 0) {
+			const transactionsResponses = await applyTransactions(
+				block.payload,
+				stateStore,
+			);
+
+			const unappliableTransactionsResponse = transactionsResponses.filter(
```

Comment:
`unappliable` is not a valid word, maybe `inapplicableTransactionsResponse` or `invalidTransactionsResponse` `CannotBeAppliedTransactions` 🤔